reframe/utility/versioning.py

```diff
@@ -0,0 +1,63 @@
+import functools
+import math
+
+
+@functools.total_ordering
+class Version:
+    def __init__(self, version):
+        try:
+            str_base, dev = version.split('-dev')
+            base = str_base.split('.')
+            try:
+                self.dev = int(dev)
+            except ValueError:
+                raise ValueError('invalid version: %s' % version)
+
+        except:
+            base = version.split('.')
+            self.dev = None
+
+        if len(base) < 2 or len(base) > 3:
+            raise ValueError('invalid version: %s' % version)
+        elif len(base) == 2:
+            base.append('0')
+
+        for i, j in enumerate(base):
+            try:
+                base[i] = int(j)
+            except ValueError:
+                raise ValueError('invalid version: %s' % version)
+
+        self.base = tuple(base)
+
+    def __eq__(self, other):
+        return self._compare(other, lambda s, o: s == o)
+
+    def __ge__(self, other):
```

Comment:
You should implement either the `__gt__` or the `__lt__`, since you already have an equality operator. Using the version value trick, the comparison should be simple. First, compare the versions and if they're not equal return the result of the comparison. If they are, then you should compare the dev numbers and return the result.